### ISSUE TITLE
#72 persistently dumps when not ran inside gdb

### ISSUE THREAD

**Issue by anonimal** [monero-project/kovri#76]
Introduced in https://github.com/monero-project/kovri/pull/72
```
terminate called after throwing an instance of 'CryptoPP::CryptoMaterial::InvalidMaterial'
  what():  CryptoMaterial: this object contains invalid values
Aborted (core dumped)
```

**Comment by anonimal** [comment]
Inaccurate title, renaming "#72 persistently dumps when not ran inside gdb". Since I can't reproduce when using gdb, the only output is what's posted above as it never gets to the logger (let alone reseed).

**Comment by anonimal** [comment]
#81 did not fix this. It's also not fixed when testing against https://github.com/majestrate/kovri/commit/8e770427516c9084104b63b320dfaff8190c0317.

gdb did comply this time:

```
terminate called after throwing an instance of 'CryptoPP::CryptoMaterial::InvalidMaterial'
  what():  CryptoMaterial: this object contains invalid values
Aborted (core dumped)
$ gdb ./kovri
(gdb) run
Starting program: /kovri/build/kovri
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[New Thread 0x7ffff25a2700 (LWP 5511)]
2016:01:25|06:31:57.411635 default:default              ERR             Can't open file ~/.kovri/router.info
*** Error in `/kovri/build/kovri': free(): corrupted unsorted chunks: 0x0000000000ec8c20 ***
======= Backtrace: =========
/usr/lib/libc.so.6(+0x72055)[0x7ffff502f055]
/usr/lib/libc.so.6(+0x779a6)[0x7ffff50349a6]
/usr/lib/libc.so.6(+0x7818e)[0x7ffff503518e]
/kovri/build/kovri(_ZN3i2p4data10RouterInfo14ReadFromBufferEb+0xd0)[0xa5ae9c]
/kovri/build/kovri(_ZN3i2p4data10RouterInfo6UpdateEPKhi+0xc9)[0xa5ab39]
/kovri/build/kovri(_ZN3i2p13RouterContext4LoadEv+0x1e2)[0xa587ca]
/kovri/build/kovri(_ZN3i2p13RouterContext4InitEv+0x2b)[0xa56e85]
/kovri/build/kovri(_ZN3i2p4util16Daemon_Singleton4InitEv+0x22)[0x93f4d2]
/kovri/build/kovri(main+0x3f)[0x93e975]
/usr/lib/libc.so.6(__libc_start_main+0xf0)[0x7ffff4fdd610]
/kovri/build/kovri(_start+0x29)[0x93e869]
======= Memory map: ========
00400000-00c68000 r-xp 00000000 fe:01 394691                             /kovri/build/kovri
00e68000-00e87000 rw-p 00868000 fe:01 394691                             /kovri/build/kovri
00e87000-00edc000 rw-p 00000000 00:00 0                                  [heap]
7fffe4000000-7fffe4021000 rw-p 00000000 00:00 0
7fffe4021000-7fffe8000000 ---p 00000000 00:00 0
7fffec000000-7fffec021000 rw-p 00000000 00:00 0
7fffec021000-7ffff0000000 ---p 00000000 00:00 0
7ffff1da2000-7ffff1da3000 ---p 00000000 00:00 0
7ffff1da3000-7ffff25a3000 rw-p 00000000 00:00 0                          [stack:5511]
7ffff25a3000-7ffff25b8000 r-xp 00000000 fe:01 792693                     /usr/lib/libz.so.1.2.8
7ffff25b8000-7ffff27b7000 ---p 00015000 fe:01 792693                     /usr/lib/libz.so.1.2.8
7ffff27b7000-7ffff27b8000 r--p 00014000 fe:01 792693                     /usr/lib/libz.so.1.2.8
7ffff27b8000-7ffff27b9000 rw-p 00015000 fe:01 792693                     /usr/lib/libz.so.1.2.8
7ffff27b9000-7ffff27bc000 r-xp 00000000 fe:01 789724                     /usr/lib/libdl-2.22.so
7ffff27bc000-7ffff29bb000 ---p 00003000 fe:01 789724                     /usr/lib/libdl-2.22.so
7ffff29bb000-7ffff29bc000 r--p 00002000 fe:01 789724                     /usr/lib/libdl-2.22.so
7ffff29bc000-7ffff29bd000 rw-p 00003000 fe:01 789724                     /usr/lib/libdl-2.22.so
7ffff29bd000-7ffff2b41000 r-xp 00000000 fe:01 817117                     /usr/lib/libicuuc.so.56.1
7ffff2b41000-7ffff2d41000 ---p 00184000 fe:01 817117                     /usr/lib/libicuuc.so.56.1
7ffff2d41000-7ffff2d51000 r--p 00184000 fe:01 817117                     /usr/lib/libicuuc.so.56.1
7ffff2d51000-7ffff2d52000 rw-p 00194000 fe:01 817117                     /usr/lib/libicuuc.so.56.1
7ffff2d52000-7ffff2d54000 rw-p 00000000 00:00 0
7ffff2d54000-7ffff2fc0000 r-xp 00000000 fe:01 817118                     /usr/lib/libicui18n.so.56.1
7ffff2fc0000-7ffff31bf000 ---p 0026c000 fe:01 817118                     /usr/lib/libicui18n.so.56.1
7ffff31bf000-7ffff31cd000 r--p 0026b000 fe:01 817118                     /usr/lib/libicui18n.so.56.1
7ffff31cd000-7ffff31cf000 rw-p 00279000 fe:01 817118                     /usr/lib/libicui18n.so.56.1
7ffff31cf000-7ffff31d0000 rw-p 00000000 00:00 0
7ffff31d0000-7ffff49b3000 r--p 00000000 fe:01 817124                     /usr/lib/libicudata.so.56.1
7ffff49b3000-7ffff4bb2000 ---p 017e3000 fe:01 817124                     /usr/lib/libicudata.so.56.1
7ffff4bb2000-7ffff4bb3000 r--p 017e2000 fe:01 817124                     /usr/lib/libicudata.so.56.1
7ffff4bb3000-7ffff4bb4000 r-xp 00000000 fe:01 818955                     /usr/lib/libboost_atomic.so.1.60.0
7ffff4bb4000-7ffff4db3000 ---p 00001000 fe:01 818955                     /usr/lib/libboost_atomic.so.1.60.0
7ffff4db3000-7ffff4db4000 r--p 00000000 fe:01 818955                     /usr/lib/libboost_atomic.so.1.60.0
7ffff4db4000-7ffff4db5000 rw-p 00001000 fe:01 818955                     /usr/lib/libboost_atomic.so.1.60.0
7ffff4db5000-7ffff4dbc000 r-xp 00000000 fe:01 789728                     /usr/lib/librt-2.22.so
7ffff4dbc000-7ffff4fbb000 ---p 00007000 fe:01 789728                     /usr/lib/librt-2.22.so
7ffff4fbb000-7ffff4fbc000 r--p 00006000 fe:01 789728                     /usr/lib/librt-2.22.so
7ffff4fbc000-7ffff4fbd000 rw-p 00007000 fe:01 789728                     /usr/lib/librt-2.22.so
7ffff4fbd000-7ffff5158000 r-xp 00000000 fe:01 789674                     /usr/lib/libc-2.22.so
7ffff5158000-7ffff5357000 ---p 0019b000 fe:01 789674                     /usr/lib/libc-2.22.so
7ffff5357000-7ffff535b000 r--p 0019a000 fe:01 789674                     /usr/lib/libc-2.22.so
7ffff535b000-7ffff535d000 rw-p 0019e000 fe:01 789674                     /usr/lib/libc-2.22.so
7ffff535d000-7ffff5361000 rw-p 00000000 00:00 0
7ffff5361000-7ffff5379000 r-xp 00000000 fe:01 789632                     /usr/lib/libpthread-2.22.so
7ffff5379000-7ffff5578000 ---p 00018000 fe:01 789632                     /usr/lib/libpthread-2.22.so
7ffff5578000-7ffff5579000 r--p 00017000 fe:01 789632                     /usr/lib/libpthread-2.22.so
7ffff5579000-7ffff557a000 rw-p 00018000 fe:01 789632                     /usr/lib/libpthread-2.22.so
7ffff557a000-7ffff557e000 rw-p 00000000 00:00 0
7ffff557e000-7ffff5594000 r-xp 00000000 fe:01 789947                     /usr/lib/libgcc_s.so.1
7ffff5594000-7ffff5793000 ---p 00016000 fe:01 789947                     /usr/lib/libgcc_s.so.1
7ffff5793000-7ffff5794000 rw-p 00015000 fe:01 789947                     /usr/lib/libgcc_s.so.1
7ffff5794000-7ffff5891000 r-xp 00000000 fe:01 789725                     /usr/lib/libm-2.22.so
7ffff5891000-7ffff5a90000 ---p 000fd000 fe:01 789725                     /usr/lib/libm-2.22.so
7ffff5a90000-7ffff5a91000 r--p 000fc000 fe:01 789725                     /usr/lib/libm-2.22.so
7ffff5a91000-7ffff5a92000 rw-p 000fd000 fe:01 789725                     /usr/lib/libm-2.22.so
7ffff5a92000-7ffff5c04000 r-xp 00000000 fe:01 805804                     /usr/lib/libstdc++.so.6.0.21
7ffff5c04000-7ffff5e04000 ---p 00172000 fe:01 805804                     /usr/lib/libstdc++.so.6.0.21
7ffff5e04000-7ffff5e0e000 r--p 00172000 fe:01 805804                     /usr/lib/libstdc++.so.6.0.21
7ffff5e0e000-7ffff5e10000 rw-p 0017c000 fe:01 805804                     /usr/lib/libstdc++.so.6.0.21
7ffff5e10000-7ffff5e14000 rw-p 00000000 00:00 0
7ffff5e14000-7ffff6062000 r-xp 00000000 fe:01 825566                     /usr/lib/libcrypto.so.1.0.0
7ffff6062000-7ffff6261000 ---p 0024e000 fe:01 825566                     /usr/lib/libcrypto.so.1.0.0
7ffff6261000-7ffff627d000 r--p 0024d000 fe:01 825566                     /usr/lib/libcrypto.so.1.0.0
7ffff627d000-7ffff6289000 rw-p 00269000 fe:01 825566                     /usr/lib/libcrypto.so.1.0.0
7ffff6289000-7ffff628c000 rw-p 00000000 00:00 0
7ffff628c000-7ffff62fc000 r-xp 00000000 fe:01 825565                     /usr/lib/libssl.so.1.0.0
7ffff62fc000-7ffff64fb000 ---p 00070000 fe:01 825565                     /usr/lib/libssl.so.1.0.0
7ffff64fb000-7ffff6500000 r--p 0006f000 fe:01 825565                     /usr/lib/libssl.so.1.0.0
7ffff6500000-7ffff6507000 rw-p 00074000 fe:01 825565                     /usr/lib/libssl.so.1.0.0
7ffff6507000-7ffff68d3000 r-xp 00000000 fe:01 824932                     /usr/lib/libcryptopp.so
7ffff68d3000-7ffff6ad3000 ---p 003cc000 fe:01 824932                     /usr/lib/libcryptopp.so
7ffff6ad3000-7ffff6b1c000 rw-p 003cc000 fe:01 824932                     /usr/lib/libcryptopp.so
7ffff6b1c000-7ffff6b20000 rw-p 00000000 00:00 0
7ffff6b20000-7ffff6c36000 r-xp 00000000 fe:01 818964                     /usr/lib/libboost_regex.so.1.60.0
7ffff6c36000-7ffff6e35000 ---p 00116000 fe:01 818964                     /usr/lib/libboost_regex.so.1.60.0
7ffff6e35000-7ffff6e3a000 r--p 00115000 fe:01 818964                     /usr/lib/libboost_regex.so.1.60.0
7ffff6e3a000-7ffff6e3c000 rw-p 0011a000 fe:01 818964                     /usr/lib/libboost_regex.so.1.60.0
7ffff6e3c000-7ffff6e3d000 rw-p 00000000 00:00 0
7ffff6e3d000-7ffff6e56000 r-xp 00000000 fe:01 818963                     /usr/lib/libboost_filesystem.so.1.60.0
7ffff6e56000-7ffff7055000 ---p 00019000 fe:01 818963                     /usr/lib/libboost_filesystem.so.1.60.0
7ffff7055000-7ffff7056000 r--p 00018000 fe:01 818963                     /usr/lib/libboost_filesystem.so.1.60.0
7ffff7056000-7ffff7057000 rw-p 00019000 fe:01 818963                     /usr/lib/libboost_filesystem.so.1.60.0
7ffff7057000-7ffff705a000 r-xp 00000000 fe:01 818956                     /usr/lib/libboost_system.so.1.60.0
7ffff705a000-7ffff7259000 ---p 00003000 fe:01 818956                     /usr/lib/libboost_system.so.1.60.0
7ffff7259000-7ffff725a000 r--p 00002000 fe:01 818956                     /usr/lib/libboost_system.so.1.60.0
7ffff725a000-7ffff725b000 rw-p 00003000 fe:01 818956                     /usr/lib/libboost_system.so.1.60.0
7ffff725b000-7ffff7282000 r-xp 00000000 fe:01 818960                     /usr/lib/libboost_thread.so.1.60.0
7ffff7282000-7ffff7481000 ---p 00027000 fe:01 818960                     /usr/lib/libboost_thread.so.1.60.0
7ffff7481000-7ffff7483000 r--p 00026000 fe:01 818960                     /usr/lib/libboost_thread.so.1.60.0
7ffff7483000-7ffff7484000 rw-p 00028000 fe:01 818960                     /usr/lib/libboost_thread.so.1.60.0
7ffff7484000-7ffff7494000 r-xp 00000000 fe:01 818959                     /usr/lib/libboost_date_time.so.1.60.0
7ffff7494000-7ffff7694000 ---p 00010000 fe:01 818959                     /usr/lib/libboost_date_time.so.1.60.0
7ffff7694000-7ffff7695000 r--p 00010000 fe:01 818959                     /usr/lib/libboost_date_time.so.1.60.0
7ffff7695000-7ffff7696000 rw-p 00011000 fe:01 818959                     /usr/lib/libboost_date_time.so.1.60.0
7ffff7696000-7ffff7713000 r-xp 00000000 fe:01 818982                     /usr/lib/libboost_program_options.so.1.60.0
7ffff7713000-7ffff7913000 ---p 0007d000 fe:01 818982                     /usr/lib/libboost_program_options.so.1.60.0
7ffff7913000-7ffff7916000 r--p 0007d000 fe:01 818982                     /usr/lib/libboost_program_options.so.1.60.0
7ffff7916000-7ffff7917000 rw-p 00080000 fe:01 818982                     /usr/lib/libboost_program_options.so.1.60.0
7ffff7917000-7ffff79cc000 r-xp 00000000 fe:01 818971                     /usr/lib/libboost_log.so.1.60.0
7ffff79cc000-7ffff7bcc000 ---p 000b5000 fe:01 818971                     /usr/lib/libboost_log.so.1.60.0
7ffff7bcc000-7ffff7bd1000 r--p 000b5000 fe:01 818971                     /usr/lib/libboost_log.so.1.60.0
7ffff7bd1000-7ffff7bd2000 rw-p 000ba000 fe:01 818971                     /usr/lib/libboost_log.so.1.60.0
7ffff7bd2000-7ffff7bd3000 rw-p 00000000 00:00 0
7ffff7bd3000-7ffff7bd9000 r-xp 00000000 fe:01 818957                     /usr/lib/libboost_chrono.so.1.60.0
7ffff7bd9000-7ffff7dd9000 ---p 00006000 fe:01 818957                     /usr/lib/libboost_chrono.so.1.60.0
7ffff7dd9000-7ffff7dda000 r--p 00006000 fe:01 818957                     /usr/lib/libboost_chrono.so.1.60.0
7ffff7dda000-7ffff7ddb000 rw-p 00007000 fe:01 818957                     /usr/lib/libboost_chrono.so.1.60.0
7ffff7ddb000-7ffff7dfd000 r-xp 00000000 fe:01 789673                     /usr/lib/ld-2.22.so
7ffff7fc3000-7ffff7fd7000 rw-p 00000000 00:00 0
7ffff7ff7000-7ffff7ff8000 rw-p 00000000 00:00 0
7ffff7ff8000-7ffff7ffa000 r--p 00000000 00:00 0                          [vvar]
7ffff7ffa000-7ffff7ffc000 r-xp 00000000 00:00 0                          [vdso]
7ffff7ffc000-7ffff7ffd000 r--p 00021000 fe:01 789673                     /usr/lib/ld-2.22.so
7ffff7ffd000-7ffff7ffe000 rw-p 00022000 fe:01 789673                     /usr/lib/ld-2.22.so
7ffff7ffe000-7ffff7fff000 rw-p 00000000 00:00 0
7ffffffde000-7ffffffff000 rw-p 00000000 00:00 0                          [stack]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]

Program received signal SIGABRT, Aborted.
0x00007ffff4ff05f8 in raise () from /usr/lib/libc.so.6
(gdb)
```

**Comment by majestrate** [comment]
try https://github.com/majestrate/kovri/commit/c45f521d1a4891be15181df26c6fc76d6410220f

**Comment by majestrate** [comment]
rather, https://github.com/majestrate/kovri/commit/b4f94eb34d6857d42ae3db93f75673d663c7e684

**Comment by anonimal** [comment]
Fixed in b4f94eb. Merged in 6d8d82e43f71440e01bd41efbe31aa2be3bb9137.
